### ISSUE TITLE
Videos without sound in 5.0.4

### ISSUE THREAD

**Issue by AaronLil** [theus/instantgram#78]
I've just tested the new version, 5.0.4, and found an issue. Videos in posts and reels are coming with no sound. In stories they are ok. Using Chrome desktop, version 108.0.5359.125 (latest).

**Comment by Xemacs64** [comment]
I'm also having the same issue, please fix this

**Comment by purple5pumpkin235** [comment]
same problem on Firefox with 5.0.4

**Comment by AaronLil** [comment]
@theus, Some useful info: the audio issue is because Instagram uses an mpeg-dash implementation and it seems that by default videos don't have audio attached to it and the mpd file has a single audio/mp4 representation which is used regardless of the quality of video displayed in on browser.

Debugging your code, I saw that it is possible to get the audio using querySelectorAll('Representation[mimeType="audio/mp4"]') on the object, but unfortunately I can't think of a solution to join video and audio in a single file without using a third party solution, like ffmpeg or youtube-dl.

Perhaps Instantgram has reached to the point where it'll be necessary to evolve beyond a bookmarlet. An extension maybe, idk.
I'm available if you need any help with this.
Best Regards.

**Comment by AnimeGeekDraws** [comment]
I JUST got 5.04 and currently dealing with the same issue. It's literally been a month since y'all posted this, so please fix it.

**Comment by Rocket3192** [comment]
Using edge, I don't have this issue

**Comment by Rocket3192** [comment]
Actually, I have it but not on all videos

**Comment by theus** [comment]
It looks like Instagram it's changing its code more frequently 😢. But I'm going to try to find a fix.

**Comment by Rocket3192** [comment]
So about that matter, i figured when it opens without sound you just need to refresh the post and try again, it's not really broken

**Comment by AaronLil** [comment]
It looks like the problem has to do with the browser/user agent. Depending on the request Instagram sends the best compatible video for that browser. Maybe that's why it works for some people and not for others.

**Comment by theus** [comment]
@AaronLil yeah, that makes sense.

**Comment by theus** [comment]
Can you guys share a post link that is getting a video without audio?

**Comment by AaronLil** [comment]
@theus, I just tested it on Chrome and Edge and _**luckily**_ I couldn't reproduce the issue.

[This video](https://www.instagram.com/reel/ClPKXV6rNUe/) certainly had the sound issue, but now it's working.
It's possible that Chrome or Instagram have updated their behaviors about this.

**Comment by theus** [comment]
@AaronLil yeah, I tested it here and it's working with sound. Should we close this issue for now? 🤔

**Comment by ghnp5** [comment]
I also had a few days/weeks of downloads without sound, but I believe this has been working well for me lately anyway.

Maybe go ahead and close this, and if we have issues again, we'll raise a new one or so :)

Thanks!!!

**Comment by Rocket3192** [comment]
@theus So I managed to generate 2 links of the same video, one containing the audio and the other one not containing it (I just refreshed the post to obtain the one containing the audio)

This one has no audio : 

https://scontent.cdninstagram.com/v/t66.30100-16/311355025_2738600936269910_8540627965462010159_n.mp4?_nc_ht=scontent-cdg2-1.cdninstagram.com&_nc_cat=102&_nc_ohc=jiWgqRrzO4sAX-PVV-K&edm=ACWDqb8BAAAA&ccb=7-5&oh=00_AfAdJmMQhknMZOhlKrvYh2WKHxquNU_Z_t7Pmx9-8D_-rg&oe=63F5C956&_nc_sid=1527a3

This one has audio : 

https://scontent.cdninstagram.com/o1/v/t16/f1/m82/86435F8C8D7EBCF78764FD5E093779BE_video_dashinit.mp4?efg=eyJxZV9ncm91cHMiOiJbXCJpZ193ZWJfZGVsaXZlcnlfdnRzX290ZlwiXSIsInZlbmNvZGVfdGFnIjoidnRzX3ZvZF91cmxnZW4uNzIwLmNsaXBzLmJhc2VsaW5lIn0&_nc_ht=scontent-cdt1-1.cdninstagram.com&_nc_cat=105&vs=1175941639587570_183274590&_nc_vs=HBksFQIYT2lnX3hwdl9yZWVsc19wZXJtYW5lbnRfcHJvZC84NjQzNUY4QzhEN0VCQ0Y3ODc2NEZENUUwOTM3NzlCRV92aWRlb19kYXNoaW5pdC5tcDQVAALIAQAVABgkR01yZzloTDNrZVQwQjVzQUFOYTAxRmYxaDE4TmJwUjFBQUFGFQICyAEAKAAYABsAFQAAJvqoqbfjg8I%2FFQIoAkMzLBdAMQAAAAAAABgSZGFzaF9iYXNlbGluZV8xX3YxEQB1%2FgcA&_nc_rid=354fa3955f&ccb=9-4&oh=00_AfASg6jzfPSvJY6-lZLbbHuZlvRhTAHrwKRbtgQqieDJZg&oe=63F58D67&_nc_sid=30a2ef

Curious thing, even the name is different
311355025_2738600936269910_8540627965462010159_n.mp4 
VS
86435F8C8D7EBCF78764FD5E093779BE_video_dashinit.mp4

VERY IMPORTANT : 
This is not the only difference, the biggest one is the resolution, one is in 1080x1920 the other one is in 720x1280 and the very sad thing is that the one with the best quality is the one MISSING the audio
It means that most likely all videos that you downloaded with sound working are actually not the best available quality


Using a famous insta downloader website it gives me the 1080x1920 file with audio and the URL is : 

https://scontent.cdninstagram.com/v/t66.30100-16/323771822_5758755634222215_6580089131176168590_n.mp4?_nc_ht=scontent.cdninstagram.com&_nc_cat=103&_nc_ohc=l5DrJAaZ304AX_YV68J&edm=APs17CUBAAAA&ccb=7-5&oh=00_AfAjD7MkbCvlJb2NhiBPFhy9SeqbRF8hyn9VPyxCE8ONlQ&oe=63F5F96E&_nc_sid=978cb9&dl=1

That's again a new name :
323771822_5758755634222215_6580089131176168590_n.mp4

PS : That's this video https://www.instagram.com/p/CoXoNgUD0TL/

The 3 files if you wanna compare them : 
https://gofile.io/d/pqG4xM

**Comment by Rocket3192** [comment]
I tested with others videos, and yep comparing with the one a download website gives me : 
instantgram game me : 
480x854 F941229048677B59AF68B88777310CA7_video_dashinit.mp4 with sound
Website gave me : 
720x1280 331154128_1202528887320202_9107745204584526551_n.mp4 with sound
It really seems like the "dashinit" file is not the maximum quality file, but it's the file that instantgram is getting, the one without audio that instantgram is randomely getting is the atual maximum resolution file, but for some reason it has no audio unlike the lower resolution dashinit one

So the goal would be to get the non dashinit one, with audio, using the gal gadot post example with the 3 different names you can try to target the same one the website is getting : 323771822_5758755634222215_6580089131176168590_n.mp4

**Comment by ramirezrick** [comment]
Still issue with sound

**Comment by theus** [comment]
Looks like Instagram sometimes (have no idea why) changes the way that videos works. It seems they split audio / video in different files and the app merges somehow like sorcery 🤯

**Comment by Floyd77** [comment]
@theus maybe you could take a look on how https://cobalt.tools/ works to extract the video with the audio.

https://github.com/imputnet/cobalt

**Comment by ghnp5** [comment]
It uses the Instagram API, with the right headers (I'd say the User-Agent is the most important thing), so that the API returns the media with video+audio together.

Note that this doesn't work for Stories.

https://github.com/imputnet/cobalt/blob/d577d5e4511bb8a21913afb7bc9acc4bb72e2f1f/api/src/processing/services/instagram.js

**Comment by AaronLil** [comment]
Switching the user agent works. For now, I'm using the extension https://webextension.org/listing/useragent-switcher.html to make this easier. Just switch the Instagram tab to IE11/Windows 10, and the video comes with audio, as it should.
